File: Data_structures/Using_perl/Ch-01/LinearSearch.pl
#!/usr/bin/env perl

use 5.014;
use strict;
use warnings;
use utf8;
binmode(STDOUT, ":utf8");

my @arr = qw/ 3 5 2 4 9 /;
my $idx;

my $sizeof =  @arr;

$idx = LSearch(\@arr, $sizeof, 4);
if ($idx == -1) {
    say "탐색 실패";
}
else {
    say "타켓 저장 인덱스: $idx";
}

$idx = LSearch(\@arr, $sizeof, 7);
if ($idx == -1) {
    say "탐색 실패";
}
else {
    say "타켓 저장 인덱스: $idx";
}

sub LSearch {
    my ($arrs, $len, $target) = @_;

    my $i = 0;
    for my $arr (@{ $arrs }) {
        if ($arr == $target) {
            return $i;
        }
        $i++;
    }

    return -1;
}

File: Data_structures/Using_C/Ch-01/LinearSearch.c
#include <stdio.h>

int
LSearch (int ar[],
         int len,
         int target)
{
  int i;

  for (i = 0; i < len; i++)
    {
      if (ar[i] == target)
        return i;
    }

  return -1;
}

int
main (int   argc,
      char *argv[])
{
  int arr[] = {3, 5, 2, 4, 9};
  int idx;

  idx = LSearch (arr, sizeof(arr) / sizeof(int), 4);
  if (idx == -1) {
    printf ("탐색 실패 \n");
  }
  else {
    printf ("타켓 저장 인덱스: %d \n", idx);
  }

  idx = LSearch(arr, sizeof (arr)/sizeof(int), 7);
  if (idx == -1) {
    printf ("탐색 실패 \n");
  }
  else {
    printf ("타켓 저장 인덱스: %d \n", idx);
  }

  return 0;
}
